src/main/java/drake/Drake.java

```diff
@@ -0,0 +1,197 @@
+package drake;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import drake.task.Deadline;
+import drake.task.Event;
+import drake.task.Task;
+import drake.task.TaskList;
+import drake.task.Todo;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+enum Command {
+    BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, INVALID, FIND;
+    public static Command fromString(String commandString) {
+        switch (commandString.toLowerCase()) {
+        case "bye":
+            return BYE;
+        case "list":
+            return LIST;
+        case "mark":
+            return MARK;
+        case "unmark":
+            return UNMARK;
+        case "todo":
+            return TODO;
+        case "deadline":
+            return DEADLINE;
+        case "event":
+            return EVENT;
+        case "delete":
+            return DELETE;
+        case "find":
+            return FIND;
+        default:
+            return INVALID;
+        }
+    }
+}
+
+/**
+ * This is the main class for the Drake application, which is a task manager program.
+ * It initializes the application, loads existing tasks, and enters a command loop that
+ * allows users to interact with their task list through various commands.
+ */
+public class Drake {
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private boolean isRunning;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/user.jpeg"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/drake.jpeg"));
```

Comment:
```suggestion
    private static final String USER_IMAGE_PATH = "/images/user.jpeg";
    private static final String DRAKE_IMAGE_PATH = "/images/drake.jpeg";

    private Image user = new Image(this.getClass().getResourceAsStream(USER_IMAGE_PATH));
    private Image duke = new Image(this.getClass().getResourceAsStream(DRAKE_IMAGE_PATH));
```

Reply:
Avoid using magic strings directly in the code. Use named constants for file paths and resource paths to improve maintainability.